### ISSUE TITLE
nerdctl build fails while building Docker/OCI dual-format tar ball 

### ISSUE THREAD

**Issue by vsiravar** [containerd/nerdctl#1804]
### Description

While building and exporting an image as an oci format tarball, nerdctl build fails with `FATA[0000] unrecognized image format`. The build is successful with `docker buildx build`

### Steps to reproduce the issue

```
% echo "FROM public.ecr.aws/docker/library/alpine:3.13" > Dockerfile
% cat Dockerfile                                                    
FROM public.ecr.aws/docker/library/alpine:3.13
% lima nerdctl build -t output:tag --output type=oci,dest=out.tar .
[+] Building 0.2s (5/5) FINISHED                                                                                                                                                                                      
 => [internal] load build definition from Dockerfile                                                                                                                                                             0.0s
 => => transferring dockerfile: 84B                                                                                                                                                                              0.0s
 => [internal] load .dockerignore                                                                                                                                                                                0.0s
 => => transferring context: 2B                                                                                                                                                                                  0.0s
 => [internal] load metadata for public.ecr.aws/docker/library/alpine:3.13                                                                                                                                       0.1s
 => CACHED [1/1] FROM public.ecr.aws/docker/library/alpine:3.13@sha256:469b6e04ee185740477efa44ed5bdd64a07bbdd6c7e5f5d169e540889597b911                                                                          0.0s
 => => resolve public.ecr.aws/docker/library/alpine:3.13@sha256:469b6e04ee185740477efa44ed5bdd64a07bbdd6c7e5f5d169e540889597b911                                                                                 0.0s
 => exporting to oci image format                                                                                                                                                                                0.1s
 => => exporting layers                                                                                                                                                                                          0.0s
 => => exporting manifest sha256:9280426892c1a5eb6882838461d4e9e5ac9a01c738ff61f031c26204d368ae49                                                                                                                0.0s
 => => exporting config sha256:e2730a754813a28b0f90c47d888aafc6c53ec1bb87da60881ee7fc4e4a99e801                                                                                                                  0.0s
 => => sending tarball                                                                                                                                                                                           0.1s
FATA[0000] unrecognized image format
```
Build works successfully with `docker buildx build`

```
 % docker buildx  build -t output:tag --output type=oci,dest=out.tar . 
[+] Building 0.5s (5/5) FINISHED                                                                                                                                                                                      
 => [internal] load .dockerignore                                                                                                                                                                                0.0s
 => => transferring context: 2B                                                                                                                                                                                  0.0s
 => [internal] load build definition from Dockerfile                                                                                                                                                             0.0s
 => => transferring dockerfile: 84B                                                                                                                                                                              0.0s
 => [internal] load metadata for public.ecr.aws/docker/library/alpine:3.13                                                                                                                                       0.4s
 => CACHED [1/1] FROM public.ecr.aws/docker/library/alpine:3.13@sha256:469b6e04ee185740477efa44ed5bdd64a07bbdd6c7e5f5d169e540889597b911                                                                          0.0s
 => => resolve public.ecr.aws/docker/library/alpine:3.13@sha256:469b6e04ee185740477efa44ed5bdd64a07bbdd6c7e5f5d169e540889597b911                                                                                 0.0s
 => exporting to oci image format                                                                                                                                                                                0.1s
 => => exporting layers                                                                                                                                                                                          0.0s
 => => exporting manifest sha256:9280426892c1a5eb6882838461d4e9e5ac9a01c738ff61f031c26204d368ae49                                                                                                                0.0s
 => => exporting config sha256:e2730a754813a28b0f90c47d888aafc6c53ec1bb87da60881ee7fc4e4a99e801                                                                                                                  0.0s
 => => sending tarball

% docker load -i out.tar 
Loaded image ID: sha256:e2730a754813a28b0f90c47d888aafc6c53ec1bb87da60881ee7fc4e4a99e801
```

### Describe the results you received and expected

I received an error instead of an image archive.

### What version of nerdctl are you using?

nerdctl version 1.0.0

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

### Host information
Darwin Kernel Version 21.6.0: Sat Jun 18 17:07:22 PDT 2022; root:xnu-8020.140.41~1/RELEASE_ARM64_T6000 arm64

**Comment by apostasie** [comment]
Hi @vsiravar 

Using main, I do not get the `unrecognized image format` error you are getting, but a mere "no image was built".

That second problem is fixed with #3388

**Comment by apostasie** [comment]
@AkihiroSuda suggesting we close this.
Fix in #3388 is obviously addressing some of the problem here, and the rest is not reproducible - OP unresponsive.
